src/main/scala/org/broadinstitute/dsde/agora/server/webservice/routes/OpenIdConnectDirectives.scala

```diff
@@ -2,25 +2,32 @@ package org.broadinstitute.dsde.agora.server.webservice.routes
 
 import akka.http.scaladsl.server.Directive1
 import akka.http.scaladsl.server.Directives._
+import org.broadinstitute.dsde.agora.server.dataaccess.SamClient
 import org.broadinstitute.dsde.agora.server.webservice.util.ImplicitMagnet
 
 import scala.concurrent.ExecutionContext
 
 
 /**
  * Directives class to retrieve user information from the OpenId Connect proxy
- * provided information in the header
+ * provided information in the header.
  */
 class OpenIdConnectDirectives extends AgoraDirectives {
+  private val serviceAccountDomain = "\\S+@\\S+\\.iam\\.gserviceaccount\\.com".r
 
-  override def commonNameFromRequest(magnet: ImplicitMagnet[ExecutionContext]): Directive1[String] = {
-    headerValueByName("OIDC_CLAIM_name")
+  private def isServiceAccount(email: String) = {
+    serviceAccountDomain.pattern.matcher(email).matches
   }
 
   override def usernameFromRequest(magnet: ImplicitMagnet[ExecutionContext]): Directive1[String] = {
-    headerValueByName("OIDC_CLAIM_email")
+    (headerValueByName("OIDC_CLAIM_email") & headerValueByName("OIDC_access_token")).tmap { case (email, token) =>
```

Comment:
Will `OIDC_access_token` always be present? `headerValueByName` will result in a `MissingHeaderRejection` if the header is missing. This is the first time Agora is expecting this header.

Reply:
it will always be present, the proxy does it and it is the same proxy as everyone else uses